test/integration/tls/tls_stack.go

```diff
@@ -0,0 +1,908 @@
+	namespace *corev1.Namespace
+
+	FrontEndVN *appmesh.VirtualNode
+	FrontEndDP *appsv1.Deployment
+
+	BackEndVN  *appmesh.VirtualNode
+	BackEndDP  *appsv1.Deployment
+	BackEndSVC *corev1.Service
+
+	BackEndVS *appmesh.VirtualService
+	BackEndVR *appmesh.VirtualRouter
+}
+
+func (s *TLSStack) DeployTLSStack(ctx context.Context, f *framework.Framework) {
+	s.createTLSStackMeshAndNamespace(ctx, f)
+	s.ServiceDiscoveryType = manifest.DNSServiceDiscovery
+	mb := &manifest.ManifestBuilder{
+		Namespace:            s.namespace.Name,
+		ServiceDiscoveryType: s.ServiceDiscoveryType,
+	}
+	s.createSecretsForTLSStack(ctx, f, mb)
+	s.createVirtualNodeResourcesForTLSStack(ctx, f, mb)
+	s.createServicesForTLSStack(ctx, f)
+	s.assignBackendVSToFrontEndVN(ctx, f)
+}
+
+func (s *TLSStack) CleanupTLSStack(ctx context.Context, f *framework.Framework) {
+	var deletionErrors []error
+	if errs := s.revokeVirtualNodeBackendAccessForTLSStack(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	if errs := s.deleteResourcesForTLSStackServices(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	if errs := s.deleteResourcesForTLSStackNodes(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	if errs := s.deleteTLSMeshAndNamespace(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	for _, err := range deletionErrors {
+		f.Logger.Error("clean up failed", zap.Error(err))
+	}
+	Expect(len(deletionErrors)).To(BeZero())
+}
+
+func (s *TLSStack) DeployPartialTLSStack(ctx context.Context, f *framework.Framework) {
+	s.createTLSStackMeshAndNamespace(ctx, f)
+	time.Sleep(30*time.Second)
+	s.ServiceDiscoveryType = manifest.DNSServiceDiscovery
+	mb := &manifest.ManifestBuilder{
+		Namespace:            s.namespace.Name,
+		ServiceDiscoveryType: s.ServiceDiscoveryType,
+	}
+	s.createSecretsForPartialTLSStack(ctx, f, mb)
+	s.createVirtualNodeResourcesForPartialTLSStack(ctx, f, mb)
+	s.createServicesForTLSStack(ctx, f)
+	s.assignBackendVSToFrontEndVN(ctx, f)
+}
+
+func (s *TLSStack) CleanupPartialTLSStack(ctx context.Context, f *framework.Framework) {
+	var deletionErrors []error
+	if errs := s.revokeVirtualNodeBackendAccessForTLSStack(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	if errs := s.deleteResourcesForTLSStackServices(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	if errs := s.deleteResourcesForPartialTLSStackNodes(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	if errs := s.deleteTLSMeshAndNamespace(ctx, f); len(errs) != 0 {
+		deletionErrors = append(deletionErrors, errs...)
+	}
+	for _, err := range deletionErrors {
+		f.Logger.Error("clean up failed", zap.Error(err))
+	}
+	Expect(len(deletionErrors)).To(BeZero())
+}
+
+func (s *TLSStack) CheckTLSBehavior(ctx context.Context, f *framework.Framework, tlsEnabled bool) {
```

Comment:
Can we please verify:
1. connectivity triggered a ssl handshake
2. refused connection when frontend client validation certs are from different CA than the backend

Reply:
Support added for a validation test case